lib/fathead/perl_doc/Parsers/parse.pl

```diff
@@ -0,0 +1,952 @@
+        };
+    }
+    return {
+        articles => \@articles,
+        aliases  => \@aliases,
+    };
+}
+
+# For docs like perlfunc, perlvar with multiple headers per entry.
+
+sub parse_variables {
+    my ($self, $dom) = @_;
+    my @mod_sections = $dom->at('a[name="General-Variables"]')
+        ->following('ul')->each;
+    ul_list_parser(
+        uls => \@mod_sections,
+        title => sub { $_[0]->find('b')->first->text . ' (variable)' },
+        aliases => sub { $_[1] =~ s/ \(variable\)//r },
+        categories => sub { ['Variables'] },
+    )->(@_);
+}
+
+#######################################################################
+#                       Normalize Parse Results                       #
+#######################################################################
+
+sub normalize_article {
+    my ($article) = @_;
+    my $text = $article->{text};
+    $text =~ s/\n/ /g;
+    return {
+        %$article,
+        text => $text,
+    };
+}
+
+sub normalize_parse_result {
+    my ($parsed) = @_;
+    $parsed->{articles} = [
+        map { normalize_article($_) } (@{$parsed->{articles}})
+    ];
+    return $parsed;
+}
+
+sub dom_for_parsing {
+    my ($url, $page) = @_;
+    my $dom = dom_for_file($page);
+    normalize_dom_links($url, $dom);
+    $dom->find('strong')->map('strip');
+    # NOTE: Once (external) parser issue is fixed, allow links in
+    # pre/code blocks
+    $dom->find('pre a, code a')->map(tag => 'span');
+    return $dom;
+}
+
+sub parse_page {
+    my ( $self, $page ) = @_;
+    my $fullpath = $self->doc_fullpath( $page->{basename} );
+    my $url = $self->doc_fullurl( $page->{filename} );
+    my $parser = $page->{parser};
+    my @parsed;
+    foreach my $parser (@{$page->{parsers}}) {
+        push @parsed, $self->$parser(dom_for_parsing($url, $fullpath));
+    }
+    foreach my $parsed (@parsed) {
+        $parsed = normalize_parse_result($parsed);
+        for my $article ( @{ $parsed->{articles} } ) {
+            my $anchored_url = $url;
+            $anchored_url .= "#" . $article->{anchor} if $article->{anchor};
+
+            $article->{url} = $anchored_url;
+            $self->article($article);
+        }
+
+        for my $alias ( @{ $parsed->{aliases} } ) {
+            $self->alias( $alias->{new}, $alias->{orig} );
+        }
+        for my $disambiguation ( @{ $parsed->{disambiguations} } ) {
+            $self->disambiguation( $disambiguation );
+        }
+    }
+}
+
+sub resolve_alias {
+    my ($self, $title) = @_;
+    my $to = $self->select(title => $title);
+    while ($to->{type} eq 'R') {
+        $to = $self->select(title => $to->{alias});
+    }
+    return $to;
+}
+
+sub resolve_aliases {
+    my ($self) = @_;
+    my %aliases = %{$self->aliases};
+    while (my ($alias, $to) = each %aliases) {
+        my @to = @$to;
+        @to == 1 and $self->insert_alias($alias, $to[0]) and next;
+        next;
+        my @articles = map { $self->resolve_alias($_) } @to;
```

Comment:
@jbarrett Well, in the IdrisDoc parser I've done it differently, see https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/lib/fathead/idris_doc/parse.pl#L581  and https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/lib/fathead/idris_doc/parse.pl#L211 where I've used the article hash as a lookup.

I think there were some multi-redirects in here as a result of the hard-coded aliases, but apart from that it should be fine to just have what's in the Idris parser.